src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/rca/temperature/NodeTemperatureRca.java

```diff
@@ -85,8 +91,13 @@ public CompactNodeTemperatureFlowUnit operate() {
       throw new IllegalStateException("One flow unit expected. Found: " + heapAllocRateFlowUnits);
     }
 
+    if (shardSizeFlowUnits.size() != 1) {
+      throw new IllegalArgumentException("One flow unit expected. Found: " + shardSizeFlowUnits);
```

Comment:
Do we want to throw an exception here if shardsize flowunit is not received ? Can we simple send out an empty flowunit if some upstream flowunits are missing ?

Reply:
That would mean we are calculating the RCA with half knowledge of metrics and calculating shard heat based on that. We are going to trigger shard movement going forward with these and this should only be done when we have full knowledge of the metrics.

Reply:
Throwing an exception is dangerous as this will mute the node and exclude it from future runs. We should not throw an exception inside operate unless it is an un-recoverable exception. 

But here we check for size to be one as this is intra-node aggregation and we don't expect multiple FlowUnits to show up just one and to @rguo-aws 's point, that can be an empty one.

Reply:
Right missed the point about muting the node. Added the empty flow unit in case no flow units are received. We are returning from NodeTemperatureRCA in the case of these empty flow units.